CITATION.cff

```diff
@@ -0,0 +1,36 @@
+cff-version: 1.2.0
+message: "Please cite this software using these metadata."
+type: software
+title: Shapely
+version: "2.0"
+# date-released: "2022-MM-DD"
+# doi: 10.5281/zenodo.#######
```

Comment:
this will be updated to the DOI that will "cite all versions" once a first release is triggered after #1379 is done.